### PR TITLE
fix(workflow): Fix non-determinism on replay when using a `patched` statement in a `condition`

### PR DESCRIPTION
**Pull request by bergundy**
Make sure not to trigger `condition`s on `notifyHasPatch` activation jobs, similar fix was applied for query jobs in #854.

This PR also uses a fixed name for the autogenerated workflow entry point to ensure that the bundle checksum is stable.

### CODE REVIEW

File: packages/test/src/test-patch-and-condition.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    });
+
+    // Trigger a workflow task that will cause replay.
+    await worker3.runUntil(async () => {
+      await handle.signal(workflows.generateCommandSignal);
+      await handle.result();
+    });
+
+    // If the workflow completes, commands are generated in the right order and it is safe to use a patched statement
+    // inside a condition.
```
[the remark this answers is not in the document]
Comment (bergundy) [reply packages/test/src/test-patch-and-condition.ts:55]: Not sure what you mean by "right branch"? Should I inspect history?

File: packages/workflow/src/worker-interface.ts [hunk 2]
```diff
@@ -222,8 +222,8 @@ export function activate(activation: coresdk.workflow_activation.WorkflowActivat
         return;
       }
       state.activator[job.variant](variant as any /* TS can't infer this type */);
-      // Only trigger conditions for non-query jobs
-      if (!job.queryWorkflow) {
+      // Only trigger conditions for non-query and non-patch jobs
+      if (!job.queryWorkflow && !job.notifyHasPatch) {
```
[the remark this answers is not in the document]
Comment (Sushisource) [reply packages/workflow/src/worker-interface.ts]: Hence predicate - just map the array and apply individually to the one job.

File: packages/test/src/test-patch-and-condition.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sushisource) [reply packages/test/src/test-patch-and-condition.ts:55]: I mean in the condition you could set something if patched was true, and return that from the workflow. Something like that.

**Review by lorensr — APPROVED** [review]
Like the comments ☺️

File: packages/workflow/src/worker-interface.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (bergundy) [reply packages/workflow/src/worker-interface.ts]: True, changed

**Pull request merged**